subgraphs/burrow/src/handlers/actions.ts

```diff
@@ -489,13 +579,32 @@ export function handleRepayment(event: EventData): void {
   const usageHourlySnapshot = getOrCreateUsageMetricsHourlySnapshot(receipt);
   const financialDailySnapshot = getOrCreateFinancialDailySnapshot(receipt);
   const account = getOrCreateAccount(account_id.toString());
-  const position = getOrCreatePosition(account, market, "BORROWER", receipt);
 
   const token = getOrCreateToken(token_id.toString());
   repay.market = market.id;
   repay.account = account.id;
   repay.amount = BigInt.fromString(amount.toString());
   repay.asset = token.id;
+
+  const counterID = account.id
+    .concat("-")
+    .concat(market.id)
+    .concat("-")
+    .concat(PositionSide.BORROWER);
+  const positionCounter = _PositionCounter.load(counterID);
+  if (!positionCounter) {
+    log.warning("Borrowing position counter {} not found in repay", [
+      counterID,
+    ]);
+    return;
+  }
+  const position = getOrCreatePosition(
```

Comment:
where is the position balance updated here?

Handling the positions in each event handler creates a lot of redundancies and makes it easier to miss things